### PR TITLE
DOC-8328: Added notes on RFC 8259

### PR DESCRIPTION
**Pull request by tonyjhillman**

### CODE REVIEW

File: modules/learn/pages/data/data.adoc [hunk 1]
```diff
@@ -9,6 +9,8 @@ Couchbase Server saves data as _items_, each of which has a _key_ and a _value_.
 Each item consists of a key and a value.
 Each key is unique within its bucket, and values can be either binary or JSON.
 
+All items are handled by Couchbase Server's xref:learn:services-and-indexes/services/data-service.adoc[Data Service]; and must be encoded as specified in https://tools.ietf.org/html/rfc8259[RFC 8259^].
```
Comment (daverigby) [inline modules/learn/pages/data/data.adoc]: So we need to caveat this - only JSON values need to confirm to RFC 8259; binary documents can be whatever the user wants.

File: modules/learn/pages/services-and-indexes/services/data-service.adoc [hunk 2]
```diff
@@ -7,6 +7,9 @@ The Data Service provides access to data.
 == Understanding the Data Service
 
 The _Data Service_ is the most fundamental of all Couchbase services, providing access to data in memory and on disk.
+
+All data-items must be encoded as specified in https://tools.ietf.org/html/rfc8259[RFC 8259^].
```
Comment (daverigby) [inline modules/learn/pages/services-and-indexes/services/data-service.adoc]: Similary here.

**Review by daverigby — CHANGES_REQUESTED** [review]

File: modules/learn/pages/services-and-indexes/services/data-service.adoc (hunk 2 above)
Comment (tonyjhillman) [reply modules/learn/pages/services-and-indexes/services/data-service.adoc]: Okay, I will add that.

Comment (tonyjhillman) [reply modules/learn/pages/services-and-indexes/services/data-service.adoc]: I've added.

**Review by daverigby — APPROVED** [review]

**Pull request merged**